sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1371 @@
+        List<TextDocumentInput> textInputs, TextAnalyticsRequestOptions options) {
+        try {
+            return withContext(context -> extractBatchKeyPhrasesWithResponse(textInputs, options, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<DocumentResultCollection<ExtractKeyPhraseResult>>> extractBatchKeyPhrasesWithResponse(
+        List<TextDocumentInput> documents, TextAnalyticsRequestOptions options, Context context) {
+        final MultiLanguageBatchInput batchInput = new MultiLanguageBatchInput()
+            .setDocuments(convertToMultiLanguageInput(documents));
+        return service.keyPhrasesWithRestResponseAsync(
+            batchInput,
+            options == null ? null : options.getModelVersion(),
+            options == null ? null : options.showStatistics(), context)
+            .doOnSubscribe(ignoredValue -> logger.info("A batch of key phrases input - {}", ignoredValue))
+            .doOnSuccess(response -> logger.info("A batch of key phrases output - {}", response.getValue()))
+            .doOnError(error -> logger.warning("Failed to key phrases - {}", error))
+            .map(response -> new SimpleResponse<>(response, toDocumentResultCollection(response.getValue())));
+    }
+
+    private DocumentResultCollection<ExtractKeyPhraseResult> toDocumentResultCollection(
+        final com.azure.ai.textanalytics.implementation.models.KeyPhraseResult keyPhraseResult) {
+        return new DocumentResultCollection<>(getKeyPhraseResults(keyPhraseResult),
+            keyPhraseResult.getModelVersion(), keyPhraseResult.getStatistics() == null ? null
+            : mapBatchStatistics(keyPhraseResult.getStatistics()));
+    }
+
+    private List<ExtractKeyPhraseResult> getKeyPhraseResults(
+        final com.azure.ai.textanalytics.implementation.models.KeyPhraseResult keyPhraseResult) {
+        List<ExtractKeyPhraseResult> validDocumentList = new ArrayList<>();
+        for (DocumentKeyPhrases documentKeyPhrases : keyPhraseResult.getDocuments()) {
+            validDocumentList.add(new ExtractKeyPhraseResult(documentKeyPhrases.getId(),
+                documentKeyPhrases.getStatistics() == null ? null
+                    : convertToTextDocumentStatistics(documentKeyPhrases.getStatistics()), null,
+                documentKeyPhrases.getKeyPhrases()));
+        }
+        List<ExtractKeyPhraseResult> errorDocumentList = new ArrayList<>();
+
+        for (DocumentError documentError : keyPhraseResult.getErrors()) {
+            final com.azure.ai.textanalytics.models.TextAnalyticsError error = convertToError(documentError.getError());
+            errorDocumentList.add(new ExtractKeyPhraseResult(documentError.getId(), null, error, null));
+        }
+        return Stream.concat(validDocumentList.stream(), errorDocumentList.stream()).collect(Collectors.toList());
```

Comment:
If we are merging all successes and errors into the same stream with the same class type why do we need to create two separate lists, get their streams, merge them, and collect them into a list, could we just create one list and feed all results into it?

Reply:
updated